x-pack/plugins/siem/public/common/components/exceptions/viewer/index.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiPanel, EuiFlexGroup, EuiCommentList, EuiAccordion, EuiFlexItem } from '@elastic/eui';
+import React, { useEffect, useState, useMemo, useCallback } from 'react';
+import styled from 'styled-components';
+
+import { ExceptionDetails } from './exception_details';
+import { ExceptionEntries } from './exception_entries';
+import { getFormattedEntries, getFormattedComments } from '../helpers';
+import { FormattedEntry, ExceptionListItemSchema } from '../types';
+
+const MyFlexItem = styled(EuiFlexItem)`
+  &.comments--show {
+    padding: 16px;
+    border-top: 1px solid #d3dae6;
+  }
+`;
+
+interface ExceptionItemProps {
+  exceptionItem: ExceptionListItemSchema;
+  handleDelete: ({ id }: { id: string }) => void;
+  handleEdit: (item: ExceptionListItemSchema) => void;
+}
+
+const ExceptionItemComponent = ({
+  exceptionItem,
+  handleDelete,
+  handleEdit,
+}: ExceptionItemProps) => {
+  const [entryItems, setEntryItems] = useState<FormattedEntry[]>([]);
+  const [showComments, setShowComments] = useState<boolean>(false);
+
+  useEffect(() => {
+    const formattedEntries = getFormattedEntries(exceptionItem.entries ?? []);
+    setEntryItems([...formattedEntries]);
+  }, [exceptionItem.entries]);
+
+  const onDelete = useCallback(() => {
+    handleDelete({ id: exceptionItem.id });
+  }, [handleDelete, exceptionItem]);
+
+  const onEdit = useCallback(() => {
+    handleEdit(exceptionItem);
+  }, [handleEdit, exceptionItem]);
+
+  const onCommentsClick = useCallback(() => {
+    setShowComments(!showComments);
+  }, [setShowComments, showComments]);
+
+  const formattedComments = useMemo(() => {
+    return getFormattedComments(exceptionItem.comments ?? []);
+  }, [exceptionItem]);
+
+  return (
+    <EuiPanel paddingSize="none">
+      <EuiFlexGroup direction="column" gutterSize="none">
+        <EuiFlexItem>
+          <EuiFlexGroup direction="row">
+            <ExceptionDetails
+              showComments={showComments}
+              exceptionItem={exceptionItem}
+              onCommentsClick={onCommentsClick}
+            />
+            <ExceptionEntries entries={entryItems} handleDelete={onDelete} handleEdit={onEdit} />
+          </EuiFlexGroup>
+        </EuiFlexItem>
+        <MyFlexItem className={showComments ? 'comments--show' : ''}>
+          <EuiAccordion
+            id="accordion--comments"
```

Comment:
Ouch...This component needs an id passed into it that it turns around and uses in the HTML?

Looks like it:
<img width="800" alt="Screen Shot 2020-06-03 at 11 00 18 AM" src="https://user-images.githubusercontent.com/1151048/83665988-7a595980-a589-11ea-91f8-c4c9f36da66b.png">


In this case since it becomes a id on a div I would bubble this id up to your component so that consumers of your component can have more than one on a page and set the id for each to a different value. Otherwise if we re-use this in more than one spot on a web page we will have a div with the same id and that could cause unexpected bugs/side effects.

Reply:
It does indeed require an `id`. I went ahead and bubbled it up to the parent. Thanks for the suggestion!